File: ruby/leap/leap.rb
class Year
  VERSION = 1

  def self.leap?(year)
    if year % 400 == 0
      yes(year)
    elsif year % 100 == 0
      no(year)
    elsif year % 4 == 0
      yes(year)
    else
      no(year)
    end
  end

  private
  def self.yes(year)
    "Yes, #{year} is a leap year"
  end

  def self.no(year)
    "No, #{year} is not a leap year"
    false
  end
end

File: javascript/leap/leap.js
export default class Leap {
  constructor (year) {
    this.year = year;
  }

  isLeap() {
    const { year } = this;
    const leap = (year % 4 === 0);
    const century = (year % 100 === 0);
    const fourHundred = (year % 400 === 0);

    return (leap && (!century || fourHundred));
  }
}
